src/librbd/io/CopyupRequest.cc

```diff
@@ -243,20 +248,40 @@ void CopyupRequest<I>::send()
 {
   m_state = STATE_READ_FROM_PARENT;
 
+  m_ictx->snap_lock.get_read();
+  m_ictx->parent_lock.get_read();
```

Comment:
nit: why not just create locker objects? I suppose we can queue a request to `op_work_queue` while holding the locks?

Reply:
It's cleaned up in the last commit